Export symbols needed for sof probe module llext

jsarha opened the pull request:
My probe llext changes still require some cleaning up, but I'll link the PR here as soon as its ready.

jsarha commented:
FYI @lyakh

teburd commented on drivers/timer/apic_tsc.c:
```diff
@@ -209,6 +209,7 @@ uint64_t sys_clock_cycle_get_64(void)
 {
 	return rdtsc();
 }
+EXPORT_SYMBOL(sys_clock_cycle_get_64);
```
I think this sort of thing would be better served in drivers/timer/export.c that optionally is built with LLEXT enabled

Since there's a few timer API symbols that we might want to export, in particular cycle_get_32 and cycle_get_64 (should be checked for availability!)

jsarha replied:
Done.

jsarha replied:
Then there is of course the third option of exporting only the intel adsp implementation, because that is all I need.

lyakh replied:
@jsarha you could also add a `__weak()` version of the function, not sure if that would be preferred.

jsarha replied:
Thanks, now there is a new version with a __weak dummy implementation of sys_clock_cycle_get_64().

teburd replied:
```c
#ifdef CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER
EXPORT_SYMBOL(sys_clock_cycle_get_64());
#endif
```

jsarha replied:
Thanks. It was not hard to find, but I did not know what to look for. Hope this is good now.

Commit:
drivers: timer: Export sys_clock_cycle_get_64() implementations

Export sys_clock_cycle_get_64() implementations to enable k_cycle_get_64()
calls from llext libraries.

Signed-off-by: Jyri Sarha <jyri.sarha@linux.intel.com>

jsarha commented on arch/xtensa/core/xtensa_asm2_export.c:
```diff
@@ -0,0 +1,9 @@
+
+#include <zephyr/arch/xtensa/arch.h>
+
+EXPORT_SYMBOL(xtensa_arch_except);
```
Sounds complicated. I just drop this part for now and leave out the single unimportant assert() from SOF probe. But I think this needs to be resolved sooner or later.

jsarha commented:
The SOF side PR requiring this change can be found here: https://github.com/thesofproject/sof/pull/9643

mmahadevan108 commented:
cc @dbaluta
